app/src/main/java/protect/card_locker/preferences/SettingsActivity.java

```diff
@@ -144,6 +148,34 @@ public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
                 }
             }
             localePreference.setEntries(entries.toArray(new CharSequence[entryValues.length]));
+            //make locale picker preference in sync with system settings
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
+                Locale chosenLocale = AppCompatDelegate.getApplicationLocales().get(0);
+                if (chosenLocale == null) {
+                    //corresponds to "System"
+                    localePreference.setValue("");
+                } else {
+                    //need to set preference's value to one of localePreference.getEntryValues() to match the locale.
+                    //Locale.toLanguageTag() theoretically should be one of the values in localePreference.getEntryValues()...
+                    //but it doesn't work for some locales. so trying something more heavyweight.
+
+                    //Obtain all locales supported by the app.
+                    List<Locale> supportedLocales = Arrays.stream(localePreference.getEntryValues())
+                            .map(Objects::toString)
+                            .map(Utils::stringToLocale)
+                            .collect(Collectors.toList());
+                    //Find the locale that's selected in settings and get its index in supported locales
+                    for (int i = 0; i < supportedLocales.size(); i++) {
+                        Locale localeToCompare = supportedLocales.get(i);
+                        //If found, set preference value to entry value at that index
+                        if (chosenLocale.equals(localeToCompare)) {
+                            localePreference.setValue(localePreference.getEntryValues()[i].toString());
+                            break;
+                        }
+                    }
+                }
+            }
+
```

Comment:
This sadly doesn't seem to work on my phone (Fairphone 3 with Android 13 (LineageOS for microG)):

:heavy_check_mark: If I set the Android per-app language to `System default`, Catima's setting switches to `System` and it uses the system locale
:x: When the Android per-app language is `System default`, trying to pick another language in Catima settings will not work, it is always forced back to `System`
:x: If I set the Android per-app language to another value, Catima's setting does not update
:x: When the Android per-app language is not `System default`, you can pick another language in Catima settings but it is not actually used in-app and Android per-app settings are not updated

Reply:
Can confirm it works now, at least on Android 13!